pep_sphinx_extensions/pep_theme/static/colour_scheme.js

```diff
@@ -0,0 +1,33 @@
+const dark = document.getElementById("css-dark");
+const pygmentsNormal = document.getElementById("pyg");
+const pygmentsDark = document.getElementById("pyg-dark");
+
+const makeLight = () => {
```

Comment:
Do we have an explicit decision on what browsers are supported for PEPs? https://caniuse.com/arrow-functions

Reply:
(Though to be clear the browsers that don't support arrow functions are really old, 2016 or earlier, and the site would probably work fine even if this file crashed, so it's not too important.)

Reply:
JS is only required for the toggle button and restoring state -- UA detection for dark mode will work without it. 

Annecdotally Sphinx is also looking to modernise its JS stack / drop old browsers (https://github.com/sphinx-doc/sphinx/issues/7405#issuecomment-971727385), so even if we wanted to support is being droped upstream.

A

Reply:
I admit I haven't tested in ancient browsers, but I expect a syntax error in JS wouldn't cancel the loading of the DOM / HTML itself. 

I tried really hard to have no JS at all (partially to avoid these questions!), but in the end I couldn't find a way to do the toggle button without it.

A

Reply:
FWIW, [CSS Variables](https://caniuse.com/css-variables) which are extensively relied upon by the stylesheet too, and for which it lacks fallbacks, are supported by about the same browsers as arrow functions, so it seems reasonable to allow them.

> I admit I haven't tested in ancient browsers, but I expect a syntax error in JS wouldn't cancel the loading of the DOM / HTML itself.

Yup, browsers are designed to be very tolerant of errors. Your browser likely runs into JS errors not infrequently, and the only indication you'll get is that something might not work, and a note in the dev console.

Reply:
I hadn't even realised CSS variables were that 'new'! (2014, but still)

A